src/StoragelessSession/Http/SessionMiddleware.php

```diff
@@ -193,54 +203,61 @@ private function parseToken(Request $request)
     }
 
     /**
-     * @param Token $token
-     *
-     * @return bool
-     */
-    private function validateToken(Token $token) : bool
-    {
-        try {
-            return $token->verify($this->signer, $this->verificationKey) && $token->validate(new ValidationData());
-        } catch (\BadMethodCallException $invalidToken) {
-            return false;
-        }
-    }
-
-    /**
      * @param Token|null $token
      *
      * @return SessionInterface
      */
     public function extractSessionContainer(Token $token = null) : SessionInterface
     {
-        return $token
-            ? DefaultSessionData::fromDecodedTokenData($token->getClaim(self::SESSION_CLAIM) ?? new \stdClass())
-            : DefaultSessionData::newEmptySession();
+        try {
+            if (null === $token || ! $token->verify($this->signer, $this->verificationKey)) {
+                return DefaultSessionData::newEmptySession();
+            }
+
+            return DefaultSessionData::fromDecodedTokenData($token->getClaim(self::SESSION_CLAIM) ?? new \stdClass());
+        } catch (\BadMethodCallException $invalidToken) {
+            return DefaultSessionData::newEmptySession();
+        }
     }
 
     /**
      * @param SessionInterface $sessionContainer
      * @param Response         $response
+     * @param Token            $token
      *
      * @return Response
      *
      * @throws \InvalidArgumentException
      */
-    private function appendToken(SessionInterface $sessionContainer, Response $response) : Response
+    private function appendToken(SessionInterface $sessionContainer, Response $response, Token $token = null) : Response
     {
         $sessionContainerChanged = $sessionContainer->hasChanged();
+        $sessionContainerIsEmpty = $sessionContainer->isEmpty();
```

Comment:
Fetching this value eagerly is not possible - if you do that, then you are initializing the session container each time